userTests/arktype/build.sh

```diff
@@ -0,0 +1,7 @@
+npm i -g pnpm
+rm -rf arktype
+git clone --depth 1 https://github.com/arktypeio/arktype.git arktype
+cd arktype
+
+pnpm i
```

Comment:
This script isn't sufficient without something that links in the copy of TS built by the pipeline; I would look at other `build.sh` files to see an example of linking.

Reply:
Ahh my bad, I was looking at userTests/typescript-eslint/build.sh but I incorrectly assumed this line was something custom to the repo I didn't have to worry about:

```ts
npx json -I -f package.json -e "this.resolutions.typescript = 'file:$TS'"
```

TBH I still have no idea what is going on there so hopefully the second half I adapted from `vue-next` works?

Reply:
That line forces `yarn` to resolve any typescript imports to the one provided by the tester, replacing what they already use to pin to a specific version across all packages: https://github.com/typescript-eslint/typescript-eslint/blob/94e809808a445dde05ab6f284dec010269482eb5/package.json#L146

Reply:
The main question with `pnpm link` is if it will apply the link to all packages in the repo; I would verify it locally just to see in your repo if every TS is replaced across the board. You might need to instead do something like `typescript-eslint` but to https://github.com/arktypeio/arktype/blob/main/pnpm-workspace.yaml since you seem to be using pnpm catalogs.

Reply:
Oh, to be clear, you can't use `resolutions` itself; that's a yarn thing. pnpm uses a different syntax in a different object.

Reply:
Sorry, I should not have pushed the intermediate commit. Based on local testing, I think this one finally works 🙏

Reply:
Uh, weird. Hopefully the version used in CI (via corepack) is new enough to have that

Reply:
I tested it locally just to check, and yeah it works.